.github/workflows/ci.yml

```diff
@@ -7,45 +7,166 @@ on:
   workflow_dispatch:
 
 jobs:
-  # Builds the library and persists it as an artifact.
-  # Later jobs can then reuse that build artifact, skipping
-  # the build step.
   build:
     runs-on: macos-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7.12
+
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
           node-version: 18
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build
+        run: pnpm build
+
+      - name: Cache build output
+        uses: actions/cache@v3
+        with:
+          path: ./lib
+          key: ${{ runner.os }}-build-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-build-
+
+  test_unit:
```

Comment:
Nitpick: let's use more human-friendly display name for this granular test jobs. I believe GitHub Actions support the `name` property on individual jobs that will be used as the display name in the GitHub UI.

```yml
test_unit:
  name: Unit tests
```

Wdyt?